sycl/doc/PreprocessorMacros.md

```diff
@@ -2,7 +2,7 @@
 
 This file describes macros that have effect on SYCL compiler and run-time.
 
-### `RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR`
+## `RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR`
```

Comment:
MyST warns when header order is violated, so I had to update this document

Reply:
Probably using headers here is not the best idea. Can we switch to the list and make macro names bold?

Reply:
Does it look better now?

Reply:
Yes. Thank you.